Add Python OMOP walkthrough (#3)
SAFEHR-data/starter-guide: 02B-Python-Omop-walkthrough-critical-care/02B-Python-Omop-walkthrough-critical-care.md

Reviewer: Can you move this to the top level of the directory so its in the same place as the R version?

Author: Done!

Reviewer: I think you've copied it, can you remove this file? and also remove the old jupyter notebook?

This document is an introductory walkthrough, demonstrating how to read into Python some OMOP data from a series of csv files, join on concept names and explore and visualise the data.

It uses synthetic data created in UCLH as a part of the  [Critical Care Health Informatics Collaborative (CCHIC) project](https://safehr-data.github.io/uclh-research-discovery/projects/uclh_cchic_s0/index.html).

# Installing & loading required packages
1. ### pip install pandas

2. ### pip install seaborn

> 3. ### pip install matplotlib

Reviewer: Happy with this documentation for now, we can tidy it up later so it we can run the notebook in CI

```python
import os
import requests
import pandas as pd
import datetime
import seaborn as sns
import matplotlib.pyplot as plt
```

# Downloading & reading in OMOP data
Here we will download & read in some UCLH critical care data stored in Github

```python

# Repository and file paths
repository = "SAFEHR-data/uclh-research-discovery"
source_path = "_projects/uclh_cchic_s0/data"

> destination_folder = "Starter_Guide_python/data"

Reviewer: Can we also move the ipyn file to /dynamic-docs/02b-python-omop-walkthrough-critical-care.ipyn please 🙏🏻 

Then for this I think you want something like this (will need to add pathlib import above) so that we can download the data relative to the notebook to the same place as the R script

```suggestion
destination_folder = Path(__file__).parent / "data"
```

        local_file.write(file_response.content)

# List all downloaded files
print("Download complete! Files in the folder:")
print(os.listdir(destination_folder))

```

    Downloading condition_occurrence.csv ******
    Downloading death.csv ******
    Downloading device_exposure.csv ******
    Downloading drug_exposure.csv ******
    Downloading measurement.csv ******
    Downloading observation.csv ******
    Downloading observation_period.csv ******
    Downloading person.csv ******
    Downloading procedure_occurrence.csv ******
    Downloading specimen.csv ******
    Downloading visit_occurrence.csv ******
    Download complete! Files in the folder:
    ['observation_period.csv', 'drug_exposure.csv', 'specimen.csv', 'death.csv', 'device_exposure.csv', 'measurement.csv', 'condition_occurrence.csv', 'visit_occurrence.csv', 'person.csv', 'observation.csv', 'procedure_occurrence.csv']

# Loading Multiple CSV Files into DataFrames
This script loads all CSV files from a specified folder into individual pandas DataFrames and stores them in a dictionary.

```python
#########Load files in Data frames w.r.t file names
# store the loaded data from each CSV file.

> df = {} 

Reviewer: I'd suggest this is called `cdm` so that it follows with the R code. If you don't like that then something like `dataframes` so that it shows that is multiple of them

> for file in os.listdir(destination_folder):
>     if file.endswith(".csv"):
>         # Extract the file name without extension
>         file_name = os.path.splitext(file)[0]
>         # Load the file into a DataFrame
>         file_path = os.path.join(destination_folder, file)
>         df[file_name] = pd.read_csv(file_path)

Reviewer: FYI if we're using path its got some nice builtins that we can use
```suggestion
for file in destination_folder.iterdir():
    if file.suffix == ".csv":
        # Extract the file name without extension
        file_name = file.stem
        # Load the file into a DataFrame
        df[file_name] = pd.read_csv(file)
        print(f"Loaded {file_name} into a DataFrame.")
```

plt.xticks(rotation=45)

# Show the plot
plt.show()
```

    
![png](output_13_0.png)
    

In the plot above bars are coloured by gender_concept_id which is the OMOP ID for gender, but we don't actually know which is which. We will look at resolving that by retrieving OMOP concept names in the next section.

# OMOP Processed Vocabularies
Repository for pre-processed vocabularies for working with the OMOP CDM at UCLH.

You can download these files directly or follow the local development instructions to use git to clone the files

## Downloading versions
You can download a specific tagged version using https. in this format, replacing the curly braced values:

https://github.com/SAFEHR-data/omop-vocabs-processed/raw/refs/tags/{tag}/{relative_path}

## Getting names for OMOP concept IDs

```python
import urllib.request

tag = "v4"
relative_path = "data/concept.parquet"
download_url = f"https://github.com/SAFEHR-data/omop-vocabs-processed/raw/refs/tags/{tag}/{relative_path}"
local_filename = "concept.parquet"

urllib.request.urlretrieve(download_url, local_filename)
```

    ('concept.parquet', <http.client.HTTPMessage at 0x14883f3e0>)

```python

> file_path = "/Users/muhammadqummerularfeen/concept.parquet" 

Reviewer: Similarly here we can use the destination_path again:
```suggestion
file_path = destination_path / "concept.parquet" 
```

visit_occurance_concept_cnt.head()
```

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Visit Concept Name</th>
      <th>Year</th>
      <th>Count</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>72</th>
      <td>Inpatient Visit</td>
      <td>2023</td>
      <td>3</td>
    </tr>
    <tr>
      <th>71</th>
      <td>Inpatient Visit</td>
      <td>2021</td>
      <td>1</td>
    </tr>
    <tr>
      <th>33</th>
      <td>Emergency Room Visit</td>
      <td>2021</td>
      <td>1</td>
    </tr>
    <tr>
      <th>70</th>
      <td>Inpatient Visit</td>
      <td>2019</td>
      <td>1</td>
    </tr>
    <tr>
      <th>32</th>
      <td>Emergency Room Visit</td>
      <td>2019</td>
      <td>1</td>
    </tr>
  </tbody>
</table>
</div>

```python
# Plot using seaborn

> plt.figure(figsize=(20, 6))

Reviewer: For this plot can we order it be year in ascending order (so RHS should have 2023)?
